sdk/core/Microsoft.Azure.Core.Spatial/src/Serialization/GeographyPointConverter.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.Spatial;
+
+namespace Azure.Core.Serialization
+{
+    /// <summary>
+    /// Converts between <see cref="GeographyPoint" /> objects and Geo-JSON points.
+    /// </summary>
+    public class GeographyPointConverter : JsonConverter<GeographyPoint>
+    {
+        // TODO: Consider reading and writing more of a GeographicPoint, but for now we only read and write what we support in OData currently.
+        private const string CoordinatesPropertyName = "coordinates";
+        private const string PointTypeName = "Point";
+        private const string TypePropertyName = "type";
+
+        private static readonly JsonEncodedText s_CoordinatesPropertyNameBytes = JsonEncodedText.Encode(CoordinatesPropertyName);
+        private static readonly JsonEncodedText s_TypePropertyNameBytes = JsonEncodedText.Encode(TypePropertyName);
+
+        /// <inheritdoc/>
+        public override GeographyPoint Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType == JsonTokenType.Null)
+            {
+                return null;
+            }
+
+            string type = default;
+            double? longitude = default;
+            double? latitude = default;
+
+            reader.Expect(JsonTokenType.StartObject);
+            while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
+            {
+                reader.Expect(JsonTokenType.PropertyName);
+                string propertyName = reader.GetString();
+
+                reader.Read();
+                if (string.Equals(TypePropertyName, propertyName, options.GetStringComparisonOrDefault()))
+                {
+                    reader.Expect(JsonTokenType.String);
+                    type = reader.GetString();
+                }
+                else if (string.Equals(CoordinatesPropertyName, propertyName, options.GetStringComparisonOrDefault()))
+                {
+                    reader.Expect(JsonTokenType.StartArray);
+
+                    // Longitude
+                    reader.Read();
+                    reader.Expect(JsonTokenType.Number);
+                    longitude = reader.GetDouble();
+
+                    // Latitude
+                    reader.Read();
+                    reader.Expect(JsonTokenType.Number);
+                    latitude = reader.GetDouble();
+
+                    // Skip the rest.
+                    do
+                    {
+                        reader.Read();
```

Comment:
https://docs.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonreader.skip?view=netcore-3.1 ?

Reply:
Thanks.

> When TokenType (originally, or after advancing) is StartObject or JsonTokenType.StartArray, the reader advances to the matching JsonTokenType.EndObject or JsonTokenType.EndArray.

I knew about the method, but Newtonsoft.Json's [lack of details](https://www.newtonsoft.com/json/help/html/M_Newtonsoft_Json_JsonReader_Skip.htm) (or anything IntelliSense doesn't already show) made me apprehensive. I'll check its implementation and perhaps use it, but it not being doc'd makes me nervous about using it. This change just ended up being equivalent, but I agree `Skip` would be better at least here.

Reply:
From 10.0.3, which we use, it seems Json.NET is the same: https://github.com/JamesNK/Newtonsoft.Json/blob/10.0.3/Src/Newtonsoft.Json/JsonReader.cs#L941-L959

Reply:
Actually, I read that wrong. For both, if the *current* token is a start token, the children are skipped. If you're in one, neither move. The code as written is correct.